### DISCUSSION TITLE
Are TSPoint column offsets in terms of bytes or code points?

### DISCUSSION THREAD

**Discussion by wiseman** [tree-sitter/tree-sitter#1286] [Q&A] [answered]
When I run `tree-sitter parse`, I see column offsets in the displayed ranges that imply the units are bytes, but when I try the Syntax Tree Playground, the offsets seem to be code points. I can't find anything in the documentation that specifies one or the other, so... which is it?

**Comment by maxbrunsfeld** [comment, marked as answer]
In the core library, columns are measured in bytes. In the JavaScript binding, those byte values are divided by two, so that the columns are measured in UTF16 code-units (notably, *not* unicode code points). This is generally what you want in JavaScript, because all of the normal JavaScript string APIs (e.g. `String.length`, `String.indexOf`, etc) work in terms of these "UTF16 code-units".

**Reply by ktraunmueller** [reply]
I couldn't find this essential bit of information in the documentation (that's why I searched the discussions). I think it would be really helpful to add this to the documentation.

**Reply by sogaiu** [reply]
Perhaps an appropriate PR would be welcome?

**Reply by ktraunmueller** [reply]
Submitted https://github.com/tree-sitter/tree-sitter/pull/3949
